### PR TITLE
MRC-143 Refactored use case of GET /config end-point

### PR DESCRIPTION
**Pull request by PavelVjalicin**

### CODE REVIEW

File: test/connectors/sharedmodel/CalculatorResultSpec.scala [hunk 1]
```diff
@@ -36,6 +36,16 @@ class CalculatorResultSpec extends AnyFreeSpec with Matchers {
     }
   }
 
+  "FlatRate" - {
+    "adjustedAugmentedProfit" - {
+      "should return sum of adjustedProfit" in {
+        flatRate.adjustedAugmentedProfit shouldBe roundUp(
+          roundUp(BigDecimal(flatRate.adjustedProfit))
```
Comment (girishkamat) [inline test/connectors/sharedmodel/CalculatorResultSpec.scala:43]: roundUp is called twice?

**Review by girishkamat — DISMISSED** [review]

File: app/views/helpers/FullResultsPageHelper.scala [hunk 2]
```diff
@@ -176,51 +184,50 @@ object FullResultsPageHelper extends ViewHelper {
...
+            )
+          )
+        ),
+        TableRow(content =
+          Text(
+            s"${if (isUpperLowerLimit) upperLimitText else lowerLimitText} × ($daysString ÷ $daysInYear $daysMsg) ÷ $pointOneCompaniesCalcText"
           )
         ),
+        TableRow(content =
+          Text(CurrencyUtils.format(if (isUpperLowerLimit) adjustedUpperLimit else marginalRate.adjustedLowerThreshold))
```
Comment (girishkamat) [inline app/views/helpers/FullResultsPageHelper.scala:206]: Can we use consistent language here? We have always used threshold, please replace limit with threshold

File: test/connectors/sharedmodel/CalculatorResultSpec.scala (hunk 1 above)
Comment (girishkamat) [inline test/connectors/sharedmodel/CalculatorResultSpec.scala:43]: Unnessary roundups here

File: app/views/helpers/FullResultsPageHelper.scala [hunk 3]
```diff
@@ -145,7 +145,17 @@ object FullResultsPageHelper extends ViewHelper {
 
     val days = marginalRate.days
 
-    val upperLimit = CurrencyUtils.format(yearConfig.upperThreshold)
+    def upperLimit = CurrencyUtils.format(yearConfig.upperThreshold)
```
Comment (girishkamat) [inline app/views/helpers/FullResultsPageHelper.scala:148]: Are these changes covered by tests?

Comment (PavelVjalicin) [reply app/views/helpers/FullResultsPageHelper.scala:148]: Whole FullResultsPageHelper is covered except for this line: throw new RuntimeException("Configuration is flat where it should be marginal")

File: app/views/helpers/FullResultsPageHelper.scala [hunk 4]
```diff
@@ -162,65 +172,65 @@ object FullResultsPageHelper extends ViewHelper {
...
-          Seq(
-            boldRow("5"),
-            TableRow(content = Text(messages("fullResultsPage.financialYear.fullCalculation"))),
-            TableRow(content = Text(s"""(${CurrencyUtils.format(adjustedUpperLimit)} - ${CurrencyUtils.format(
-                taxableProfitIncludingDistributions
-              )}) × (${CurrencyUtils.format(
-                marginalRate.adjustedProfit
-              )} ÷ ${CurrencyUtils.format(taxableProfitIncludingDistributions)}) × ($fraction)""")),
-            TableRow(content = Text(CurrencyUtils.format(ResultsPageHelper.marginalRelief(marginalRate))))
+    val isUpperLowerLimit = taxableProfitIncludingDistributions >= marginalRate.adjustedLowerThreshold
```
Comment (girishkamat) [inline app/views/helpers/FullResultsPageHelper.scala:186]: isProfitsAboveLowerThreshold?

Also, should this be >= or >

Comment (PavelVjalicin) [reply app/views/helpers/FullResultsPageHelper.scala:186]: After re reading the ticket I think it should be > yes.

**Review by girishkamat — APPROVED** [review]

**Pull request merged**